### PR TITLE
INFRA-28793: Re-factor how we handle instrumentation

### PR DESCRIPTION
**Pull request by nabadger**
tldr:

Move away frrom using the operator because:
- Using the operator adds complications around auto-instrumentation for python apps, as it manipulates the `PYTHONPATH`, which is risky if the apps are not expecting it (leading to failed pods). 
- Using the operator just to inject env-vars via the `Instrumentation` resource creates issues because it only injects when pods are created (not when we change values in the resource itself).

---

New config will be something like:
```
otel:
  exporter:
    endpoint: "http://..."
  sampler:
    type: parentbased_always_on
    arg: "" # optional
  extraEnv: []
  python:
    enabled: true|false
    excludedUrls: "/healthy/...."
    injectSidecar: true|false # not yet impl
    extraEnv: 
      - name: ...
        value: ...
  java: 
    ...
```

---

### Changed
- Configure open-telemetry for apps using env-vars rather than the operator
- Resource attributes do not need to pass in all pod-info (derived by prom scrape config)

### Removed
- Removed the open-telemetry `Instrumentation` resource

## [v3.61.6] - 2022-03-08

### CODE REVIEW

File: charts/standard-application-stack/values.yaml [hunk 1]
```diff
@@ -427,136 +427,41 @@ persistentVolumes:
...
+    type: parentbased_always_on
+
+    # -- Configures OTEL_TRACES_SAMPLER_ARG
+    arg: ""
+
+  # -- Additional otel (generic) vars to add to the pod
+  extraEnv: []
+
+  # -- Enable/disable injecting auto-instrumentation sidecar
+  inject: false
```
Comment (jtdoepke) [inline charts/standard-application-stack/values.yaml:448]: Do you intend to implement this in this PR? If not, I'd remove this for now.

Comment (nabadger) [reply charts/standard-application-stack/values.yaml:448]: Not yet - in a future MR. Will remove for now.

File: charts/standard-application-stack/Chart.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ type: application
 # This is the chart version. This version number should be incremented each time you make changes
 # to the chart and its templates, including the app version.
 # Versions are expected to follow Semantic Versioning (https://semver.org/)
-version: 3.61.5
+version: 3.62.0
```
Comment (jtdoepke) [inline charts/standard-application-stack/Chart.yaml:18]: This should probably be `v4.0.0` because we're removing functionality.

**Review by adamhf — APPROVED** [review]
